node-red-project/flows.json

```diff
@@ -9372,5 +9380,521 @@
+        "name": "Uptime",
+        "rules": [
+            {
+                "t": "set",
+                "p": "topic",
+                "pt": "msg",
+                "to": "metrics_Uptime",
+                "tot": "str"
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 470,
+        "y": 200,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "882369d2082717a5",
+        "type": "change",
+        "z": "f410f6aac9d1a3fc",
+        "name": "CPUs",
+        "rules": [
+            {
+                "t": "set",
+                "p": "topic",
+                "pt": "msg",
+                "to": "metrics_CPUx",
+                "tot": "str"
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 470,
+        "y": 260,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "b6b8a1a7c08eb0ac",
+        "type": "change",
+        "z": "f410f6aac9d1a3fc",
+        "name": "Loadavg",
+        "rules": [
+            {
+                "t": "set",
+                "p": "topic",
+                "pt": "msg",
+                "to": "metrics_Loadavg",
+                "tot": "str"
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 480,
+        "y": 320,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "976ec527b724bd3d",
+        "type": "change",
+        "z": "f410f6aac9d1a3fc",
+        "name": "Memory",
+        "rules": [
+            {
+                "t": "set",
+                "p": "topic",
+                "pt": "msg",
+                "to": "metrics_Memory",
+                "tot": "str"
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 480,
+        "y": 380,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "6881f1dd8a4805af",
+        "type": "change",
+        "z": "f410f6aac9d1a3fc",
+        "name": "NetworkIntf",
+        "rules": [
+            {
+                "t": "set",
+                "p": "topic",
+                "pt": "msg",
+                "to": "metrics_NetworkIntf",
+                "tot": "str"
+            }
+        ],
+        "action": "",
+        "property": "",
+        "from": "",
+        "to": "",
+        "reg": false,
+        "x": 490,
+        "y": 440,
+        "wires": [
+            [
+                "c497cfedf6aed533"
+            ]
+        ]
+    },
+    {
+        "id": "f894a3a8017a7de4",
```

Comment:
Instead of setting up another HTTP request, send the payload through the `thingsboard-telemetry` flow using `link in`/`link out` nodes, similar to the other ones there:
- Put the link out after the `Compose Metrics Msg 8`
- Move the `format metrics payload` node to that flow.

Slot it in here:
![image](https://github.com/user-attachments/assets/d6413f15-f7f6-4fb2-87f0-ccec90509522)

Reply:
The reason I did not do this in the first place is that the ` tb telemetry http api ` node is hardcoded to send telemetry, but the `metrics flow` is sending metrics.

To reuse the logic for sending messages, I would:

- Add `type` as another parameter used by the ` tb telemetry http api` node to construct the URL
- set this `type` parameter in a new node that is outside the `TB HTTP API` group
- change all the links on the `thingsboard-telemetry` page that currently reach the `TB HTTP API` group to instead link to this new node

There would still be a `format metrics payload` node on the `metrics` flow, but it would also set the new `type` field before linking to the `TP HTTP API` node.

Do you agree?

Reply:
In Metrics you are using the same endpoint:  `${serverUrl}:${serverPort}/api/v1/${accessToken}/telemetry` as _thingsboard-telemetry_ flow:  `${serverUrl}:${serverPort}/api/v1/${accessToken}/telemetry`.  In fact, you set up the code in both flows, and I ran with that.

Run all that through the same http node so we can properly track using the rate limiting node.

Semantically _telemetry_ and _metrics_ are interchangeable. ThingsBoard uses the term **telemetry**, and it goes to the same endpoint.

Reply:
OK, you're right.  I will fix the metrics flow.

Reply:
Done.